views/js/productListingComments.js

```diff
@@ -87,7 +87,7 @@ var productListingComments = (function () {
         });
 
 
-        var IDsArray = Object.keys(seenIds);
+        var IDsArray = Object.keys(seenIds).filter(e => e !== 'undefined');
```

Comment:
```suggestion
        var IDsArray = Object.keys(seenIds).filter(e => typeof e !== 'undefined');
```

Reply:
when I checked e was a string, so typeof e was returning string

Reply:
Well, then this come from the backend